src/Chip/Chip.js

```diff
@@ -0,0 +1,172 @@
+
+export const styleSheet = createStyleSheet('Chip', (theme) => {
+  const { palette, shadows, transitions } = theme;
+  const height = 32;
+  const backgroundColor = emphasize(palette.background.default, 0.12);
+  const deleteIconColor = fade(palette.text.primary, 0.26);
+  return {
+    root: {
+      fontSize: 13,
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height,
+      color: palette.getContrastText(backgroundColor),
+      backgroundColor,
+      borderRadius: height / 2,
+      whiteSpace: 'nowrap',
+      width: 'fit-content',
+      transition: transitions.create(),
+      cursor: 'default', //  set cursor to default label to inherit - clickable class overrides this
+    },
+    clickable: {
+      '&:hover, &:focus': {
+        backgroundColor: emphasize(backgroundColor, 0.08),
+      },
+      '&:active': {
+        boxShadow: shadows[1],
+        backgroundColor: emphasize(backgroundColor, 0.12),
+      },
+      cursor: 'pointer',
+    },
+    avatar: {
+      marginRight: -4,
+      width: 32,
+      height: 32,
+      fontSize: 16,
+    },
+    avatarIcon: {
+      width: 19,
+      height: 19,
+    },
+    label: {
+      paddingLeft: 12,
+      paddingRight: 12,
+      userSelect: 'none',
+      whiteSpace: 'nowrap',
+      cursor: 'inherit',
+    },
+    deleteIcon: {
+      color: deleteIconColor,
+      '&:hover': {
+        color: fade(deleteIconColor, 0.4),
+      },
+      cursor: 'pointer',
+      margin: '0 4px 0 -8px',
+    },
+  };
+});
+
+/**
+ * Chips represent complex entities in small blocks, such as a contact.
+ *
+ * ```jsx
+ * <Chip>
+ *   <Avatar>
+ *   Label text
+ * </Chip>
+ * ```
+ */
+export default function Chip(props, context) {
+  const {
+    avatar: avatarProp,
+    className: classNameProp,
+    deleteIconClassName: deleteIconClassNameProp,
+    label,
+    labelClassName: labelClassNameProp,
+    onRequestDelete,
+    onClick,
+    ...other
+  } = props;
+
+  const classes = context.styleManager.render(styleSheet);
+  const className = classNames(classes.root, { [classes.clickable]: onClick }, classNameProp);
+  const labelClassName = classNames(classes.label, labelClassNameProp);
+
+
+  let deleteIconClassName;
+  let deleteIcon = null;
+  if (onRequestDelete) {
+    deleteIconClassName = classNames(classes.deleteIcon, deleteIconClassNameProp);
+    deleteIcon =
+      (<DeleteIcon
+        className={deleteIconClassName}
+        onClick={onRequestDelete}
+      />);
+  }
+
+  let avatar = null;
+  if (React.isValidElement(avatarProp)) {
+    const avatarClassName = classNames(classes.avatar, avatarProp.props.className);
+
+    let avatarChildren = avatarProp.props.children;
+    if (avatarChildren) {
+      // If Avatar has an icon, style it.
+      if (React.isValidElement(avatarChildren)) {
+        const avatarIconClassName = classNames(classes.avatarIcon, avatarChildren.props.className);
```

Comment:
That's how I did it in a previous commit  (well, `iconClassName`) - see my comment above:

> In doing so, I also removed the iconClassName prop that I previously added to Avatar.

This makes Avatar simpler, and removes an otherwise redundant prop from Avatar (users can style the Avatar icon directly), but it does make the Chip code more complex. I'm open to your suggestion on what's preferable here.

I think making Avatar `childrenClassName` hidden in the docs is probably the best compromise.